Coroutine.ttslua

```diff
@@ -1,16 +1,28 @@
 ---@class ge_tts__Coroutine
 local Coroutine = {}
 
---- Yields from the current coroutine, which will later be resumed once a condition is met.
+--- Yields from the current coroutine, to be resumed once a condition is met or an optional timeout limit is reached.
 ---@param condition fun(): boolean @Return true when the current coroutine should be resumed.
-function Coroutine.yieldCondition(condition)
+---@param timeout number|nil @Timeout in seconds (optional).
+---@return boolean @Return whether the condition was met or not.
+---@overload fun(condition: fun():boolean):boolean
+function Coroutine.yieldCondition(condition, timeout)
     local co = coroutine.running()
 
-    Wait.condition(function()
-        coroutine.resume(co)
-    end, condition)
+    local timeOutFunc
+    if type(timeout) == "number" then
+        timeOutFunc = function()
+            coroutine.resume(co, false)
+        end
+    end
 
-    coroutine.yield()
+    local function resume()
+        coroutine.resume(co, true)
+    end
+
+    Wait.condition(resume, condition, timeout, timeOutFunc)
+
+    return coroutine.yield()
```

Comment:
i don't understand this one, is this because something could preempt the scheduled resume (since we already scheduled a boolean to be passed into the resume)

Reply:
Oh, nah. Nothing that fancy. `coroutine.yield` just has the EmmyLua doc type:

```lua
---@vararg any
---@return any...
function coroutine.yield(...) end
```

The return type is `any...` so, which could be _no_ return value at all (e.g. `nil`).

So when we call this function EmmyLua is basically going:

> Hmm, we've said this function returns a `boolean`, but `coroutine.yield()` may actually return `nil`. Nope, they're not compatible types.

_Technically_ EmmyLua is actually correct here as well. Because at some other stage, another piece of code _could_ have grabbed a hold of this coroutine (Lua `thread`) and call `coroutine.resume(ourCoroutine)` i.e. We _may_ actually get `nil` here, or well, any value at all!

However, in this particular case, I think it's fair to assume users won't do bonkers stuff like that, and thus assume that it's only us calling `coroutine.resume`. In which case, we _know_ we're resuming with either `true` or `false`, so we can "safely" (with the aforementioned caveats) cast to `boolean`.